src/cache/mod.rs

```diff
@@ -0,0 +1,122 @@
+use std::collections::hash_map::HashMap;
+
+use crate::protocol::properties::ContentChange;
+use std::sync::{Arc, Mutex};
+
+lazy_static! {
```

Comment:
I think we shouldn't use any singleton model. Instead we should add the cache to any depend handler. 

Here are some explanations about singleton model.
https://www.davidtanzer.net/david's%20blog/2016/03/14/6-reasons-why-you-should-avoid-singletons.html
https://stackoverflow.com/questions/137975/what-is-so-bad-about-singletons

Reply:
Normally I would agree with you but for a global cache I prefer this pattern rather than passing around an cache instance.

Reply:
Actually, it's the same reason, we should never use any global cache. If we make something global hidden in the code, it should be really bad to debug or test.

Reply:
but a global cache is what we need to implement this feature

Reply:
The cache itself is a struct which we could open up to easily test, the global functions merely call into an instance of that struct.

Reply:
I agree that singletons are usually an anti-pattern, but I think it's okay in this situation. Caches are literally singletons.

If we need/want to use a DI pattern later for testing, it'd be fairly trivial to do so.